src/lib/features/feature-toggle/feature-toggle-strategies-store.ts

```diff
@@ -559,29 +561,39 @@ class FeatureStrategiesStore implements IFeatureStrategiesStore {
         if (projectId) {
             query = query.where({ project: projectId });
         }
-
-        if (queryString?.trim()) {
+        const hasQueryString = Boolean(queryString?.trim());
+        const hasHalfTag = normalizedHalfTag && normalizedHalfTag.length > 0;
+        if (hasQueryString || hasHalfTag) {
+            const tagQuery = this.db.from('feature_tag').select('feature_name');
             // todo: we can run a cheaper query when no colon is detected
-            const tagQuery = this.db
-                .from('feature_tag')
-                .select('feature_name')
-                .whereRaw("(?? || ':' || ??) LIKE ?", [
+            if (hasQueryString) {
+                tagQuery.whereRaw("(?? || ':' || ??) ILIKE ?", [
                     'tag_type',
                     'tag_value',
                     `%${queryString}%`,
                 ]);
+            }
+            if (hasHalfTag) {
+                const tagParameter = normalizedHalfTag.map((tag) => `%${tag}%`);
+                tagQuery.orWhereRaw(
+                    `(?? || ':' || ??) ILIKE ANY (ARRAY[${tagParameter
+                        .map(() => '?')
+                        .join(',')}])`,
+                    ['tag_type', 'tag_value', ...tagParameter],
+                );
```

Comment:
This is difficult to understand from a glance

Reply:
I will be simplifying this in following PR, because of performance reasons.